### ISSUE TITLE
Use of bias term in tensorflow.keras conv3Dtranspose layer breaks if layer size is not defined

### ISSUE THREAD

**Issue by str4w** [tensorflow/tensorflow#22771]
### System information

    Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
    OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows, Colab
    Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: NA
    TensorFlow installed from (source or binary): binary
    TensorFlow version (use command below): 1.10 / 1.11
    Python version: 3.6.x
    Bazel version (if compiling from source): NA
    GCC/Compiler version (if compiling from source): NA
    CUDA/cuDNN version: n/a
    GPU model and memory: n/a
    Exact command to reproduce: see code below

### Describe the problem
If the input layer size is None for a conv3Dtranspose layer the calculation of the bias term emits an error because it reshapes the tensor to be multiplied by the bias, even when just initializing with no input.  Note that this is supported for the 2D layer.

### Source code / logs
https://colab.research.google.com/gist/str4w/d60705ac54c5574f67ff2ec15cf89b42/conv_transpose_problem.ipynb

```
# use these to show tensorflow.keras issue
import tensorflow as tf
import tensorflow.keras as keras
import tensorflow.keras.layers as kl
import tensorflow.keras.models as km
import tensorflow.keras.backend as K
print(tf.__version__)
print(keras.__version__)

import numpy as np


# In 2d, the conv2dtranspose layer works fine with bias.

inp=kl.Input(shape=(None,None,1))
x=kl.Conv2D(1,kernel_size=5,padding='same')(inp)
x=kl.MaxPooling2D(pool_size=2)(x)
x=tf.layers.Conv2DTranspose(1,kernel_size=1,strides=2,use_bias=True)(x)
#x=kl.Conv2DTranspose(1,kernel_size=1,strides=2)(x)
model=km.Model(inputs=inp,outputs=x)
model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=keras.optimizers.Adadelta(),
              metrics=['accuracy'])
model.summary()
data2D=np.random.random_sample((1,42,74,1))
Z2=model.predict(data2D)
assert(Z2.shape == data2D.shape)


# in 3d, the bias term has issues
# Setting use_bias to false will allow this to pass
inp=kl.Input(shape=(None,None,None,1))
x=kl.Conv3D(1,kernel_size=5,padding='same')(inp)
x=kl.MaxPooling3D(pool_size=2)(x)
x=kl.Conv3DTranspose(1,kernel_size=1,strides=2,use_bias=True)(x)
#x=kl.Conv3DTranspose(1,kernel_size=1,strides=2)(x)
model=km.Model(inputs=inp,outputs=x)
model.compile(loss=keras.losses.categorical_crossentropy,
              optimizer=keras.optimizers.Adadelta(),
              metrics=['accuracy'])
model.summary()

data3D=np.random.random_sample((1,42,74,34,1))
Z3=model.predict(data3D)
assert(Z3.shape == data3D.shape)
```
Output:
```
1.11.0
2.1.6-tf
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
=================================================================
input_1 (InputLayer)         (None, None, None, 1)     0         
_________________________________________________________________
conv2d (Conv2D)              (None, None, None, 1)     26        
_________________________________________________________________
max_pooling2d (MaxPooling2D) (None, None, None, 1)     0         
_________________________________________________________________
conv2d_transpose_1 (Conv2DTr (None, None, None, 1)     2         
=================================================================
Total params: 28
Trainable params: 28
Non-trainable params: 0
_________________________________________________________________

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-ff267c7f58e9> in <module>()
     32 x=kl.Conv3D(1,kernel_size=5,padding='same')(inp)
     33 x=kl.MaxPooling3D(pool_size=2)(x)
---> 34 x=kl.Conv3DTranspose(1,kernel_size=1,strides=2,use_bias=True)(x)
     35 #x=kl.Conv3DTranspose(1,kernel_size=1,strides=2)(x)
     36 model=km.Model(inputs=inp,outputs=x)

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, inputs, *args, **kwargs)
    767 
    768       if not in_deferred_mode:
--> 769         outputs = self.call(inputs, *args, **kwargs)
    770         if outputs is None:
    771           raise ValueError('A layer\'s `call` method should return a Tensor '

/usr/local/lib/python3.6/dist-packages/tensorflow/python/keras/layers/convolutional.py in call(self, inputs)
   1064       else:
   1065         outputs_4d = array_ops.reshape(outputs, [
-> 1066             outputs_shape[0], outputs_shape[1] * outputs_shape[2],
   1067             outputs_shape[3], outputs_shape[4]
   1068         ])

TypeError: unsupported operand type(s) for *: 'NoneType' and 'NoneType'
```

Pinging: @nuance-research

**Comment by facaiy** [comment]
@yongtang I think we can fix the issue easily after #22127 is merged . How do you think?

**Comment by yongtang** [comment]
@facaiy Yes I rebased #22127 to resolve the merge conflict. Hopefully it will be merged soon.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @facaiy: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @facaiy: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @facaiy: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by facaiy** [comment]
I think the issue has been fixed in #23004. Feel free to reopen it if have any additional questions. Thanks.

**Comment by Meiyd** [comment]
Hi, I have a got a new error like this:
My codes are:
def autoencoder1(file_length):


    input_img = keras.layers.Input( shape=( None, len(params2), len(params2), 1 ) ) # adapt this if using `channels_first` image data format

    conv1 = keras.layers.Conv3D(filters=32, kernel_size=(1, 3, 3), strides=(1, 1, 1), activation='selu', padding='same', data_format='channels_last', name='conv1')(input_img)
    conv2 = keras.layers.Conv3D(filters=64, kernel_size=(1, 3, 3), strides=(1, 2, 2), activation='selu', padding='same', data_format='channels_last', name='conv2')(conv1)
    conv3 = keras.layers.Conv3D(filters=128,kernel_size=(1, 3, 3), strides=(1, 2, 2), activation='selu', padding='same', data_format='channels_last', name='conv3')(conv2)
    conv4 = keras.layers.Conv3D(filters=256,kernel_size=(1, 3, 3), strides=(1, 2, 2), activation='selu', padding='same', data_format='channels_last', name='conv4')(conv3)

    convlstm1 = keras.layers.ConvLSTM2D(filters=32, return_sequences=True, kernel_size=(3, 3), strides=(1, 1), activation='selu', padding='same', data_format='channels_last', name='convlstm1')(conv1)
    convlstm2 = keras.layers.ConvLSTM2D(filters=64, return_sequences=True, kernel_size=(3, 3), strides=(1, 1), activation='selu', padding='same', data_format='channels_last', name='convlstm2')(conv2)
    convlstm3 = keras.layers.ConvLSTM2D(filters=128,return_sequences=True, kernel_size=(3, 3), strides=(1, 1), activation='selu', padding='same', data_format='channels_last', name='convlstm3')(conv3)
    convlstm4 = keras.layers.ConvLSTM2D(filters=256,return_sequences=True, kernel_size=(3, 3), strides=(1, 1), activation='selu', padding='same', data_format='channels_last', name='convlstm4')(conv4)
    
    deconv4 = keras.layers.Conv3DTranspose(filters=128, kernel_size=(1, 2, 2), strides=(1, 2, 2), activation='selu', padding='valid', output_padding=(0, -1, -1), data_format='channels_last', name='deconv4')(convlstm4)
    concat4 = keras.layers.Concatenate(axis=4, name='concat4')([convlstm3, deconv4])
    deconv3 = keras.layers.Conv3DTranspose(filters=64 , kernel_size=(1, 2, 2), strides=(1, 2, 2), activation='selu', padding='valid', output_padding=(0, -1, -1), data_format='channels_last', name='deconv3')(concat4)
    concat3 = keras.layers.Concatenate(axis=4, name='concat3')([convlstm2, deconv3])
    deconv2 = keras.layers.Conv3DTranspose(filters=32 , kernel_size=(1, 3, 3), strides=(1, 2, 2), activation='selu', padding='same' , data_format='channels_last', name='deconv2')(concat3)
    concat2 = keras.layers.Concatenate(axis=4, name='concat2')([convlstm1, deconv2])
    deconv1 = keras.layers.Conv3DTranspose(filters=1  , kernel_size=(1, 3, 3), strides=(1, 1, 1), activation='selu', padding='same' , data_format='channels_last', name='deconv1')(concat2)

    autoencoder = keras.models.Model(input_img, deconv1)
    autoencoder.compile(optimizer='adadelta', loss='binary_crossentropy')
    autoencoder.summary()
    
    return autoencoder

autoencoder1(1700)

and here's the error message:
TypeError                                 Traceback (most recent call last)
<ipython-input-152-83622ca7a77b> in <module>
     28     return autoencoder
     29 
---> 30 autoencoder1(1700)

<ipython-input-152-83622ca7a77b> in autoencoder1(file_length)
     14     convlstm4 = keras.layers.ConvLSTM2D(filters=256,return_sequences=True, kernel_size=(3, 3), strides=(1, 1), activation='selu', padding='same', data_format='channels_last', name='convlstm4')(conv4)
     15 
---> 16     deconv4 = keras.layers.Conv3DTranspose(filters=128, kernel_size=(1, 2, 2), strides=(1, 2, 2), activation='selu', padding='valid', output_padding=(0, -1, -1), data_format='channels_last', name='deconv4')(convlstm4)
     17     concat4 = keras.layers.Concatenate(axis=4, name='concat4')([convlstm3, deconv4])
     18     deconv3 = keras.layers.Conv3DTranspose(filters=64 , kernel_size=(1, 2, 2), strides=(1, 2, 2), activation='selu', padding='valid', output_padding=(0, -1, -1), data_format='channels_last', name='deconv3')(concat4)

~/anaconda3/lib/python3.7/site-packages/tensorflow/python/keras/engine/base_layer.py in __call__(self, inputs, *args, **kwargs)
    552             # In graph mode, failure to build the layer's graph
    553             # implies a user-side bug. We don't catch exceptions.
--> 554             outputs = self.call(inputs, *args, **kwargs)
    555           else:
    556             try:

~/anaconda3/lib/python3.7/site-packages/tensorflow/python/keras/layers/convolutional.py in call(self, inputs)
   1138       else:
   1139         outputs_4d = array_ops.reshape(outputs, [
-> 1140             outputs_shape[0], outputs_shape[1] * outputs_shape[2],
   1141             outputs_shape[3], outputs_shape[4]
   1142         ])

TypeError: unsupported operand type(s) for *: 'NoneType' and 'int'

It's April 2019 and I think there are still problems in this way.

**Comment by facaiy** [comment]
Hi, please create a new issue and fill out all necessary information?
